[Bug]: Some OS constants are redefined in `mqtt_wss_client.c`: `warning: "NSEC_PER_USEC" redefined` etc.

### Bug description

The code redefines several OS constants:
```
mqtt_wss_client.c: warning: "NSEC_PER_USEC" redefined
```
etc.

### Expected behavior

It may not be safe to redefine OS constants. Custom variables may be preferable.

### Steps to reproduce

Run the build on a similar configuration.

### Installation method

from source

### System info

```shell
$ sw_vers
ProductName:    Mac OS X
ProductVersion:    10.6.8
BuildVersion:    10K549
$ uname -a
Darwin Sergey-Fedorovs-Mac.local 10.8.0 Darwin Kernel Version 10.8.0: Tue Jun  7 16:34:44 PDT 2011; root:xnu-1504.15.3~1/RELEASE_PPC Power Macintosh
```

### Netdata build info

```shell
$ netdata -W buildinfo 
time=2025-07-29T21:55:27.666+08:00 comm=netdata source=daemon level=notice errno="2, No such file or directory" tid=18446744072100750880  msg="CONFIG: cannot load user config '/opt/local/etc/netdata/stream.conf'. Will try stock config."
machdep: class is not implemented
/opt/local/libexec/netdata/plugins.d/system-info.sh: line 405: * 1024: syntax error: operand expected (error token is "* 1024")
time=2025-07-29T21:55:28.854+08:00 comm=spawn-unnamed source=collector level=warning tid=18446744072100750880  msg="SPAWN SERVER: child with pid 11687 (request 1) exited with exit code 1: /bin/sh -c /opt/local/libexec/netdata/plugins.d/system-info.sh"
Packaging:
    Netdata Version ____________________________________________ : v2.6.1
    Installation Type __________________________________________ : custom
    Package Architecture _______________________________________ : unknown
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ : cmake -DCMAKE_BUILD_TYPE=MacPorts -DCMAKE_C_STANDARD=11 -DCMAKE_CXX_STANDARD=14 -DBUILD_SHARED_LIBS=ON -DCMAKE_C_FLAGS='-pipe -I/opt/local/libexec/openssl3/include -O2 -DNDEBUG -I/opt/local/libexec/openssl3/include -isystem/opt/local/include/LegacySupport -isystem/opt/local/include -fexceptions -fstack-protector-strong -D_FORTIFY_SOURCE=3 -fstack-clash-protection -ffunction-sections -fdata-sections -Wno-builtin-macro-redefined -fexceptions -fno-omit-frame-pointer -funwind-tables -fasynchronous-unwind-tables' -DCMAKE_CXX_FLAGS='-pipe -I/opt/local/libexec/openssl3/include -O2 -DNDEBUG -I/opt/local/libexec/openssl3/include -isystem/opt/local/include/LegacySupport -isystem/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -pipe -I/opt/local/libexec/openssl3/include -O2 -DNDEBUG -I/opt/local/libexec/openssl3/include -isystem/opt/local/include/LegacySupport -isystem/opt/local/include -fexceptions -fstack-protector-strong -D_FORTIFY_SOURCE=3 -fstack-clash-protection -ffunction-sections -fdata-sections -Wno-builtin-macro-redefined -fexceptions -fno-omit-frame-pointer -funwind-tables -fasynchronous-unwind-tables' -DCMAKE_COMPILE_DEFINITIONS='' -DCMAKE_EXE_LINKER_FLAGS='-L/opt/local/libexec/openssl3/lib -Wl,-headerpad_max_install_names -latomic -L/opt/local/lib -lMacportsLegacySupport -fexceptions -fstack-protector-strong -D_FORTIFY_SOURCE=3 -fstack-clash-protection -ffunction-sections -fdata-sections -Wno-builtin-macro-redefined -fexceptions -rdynamic' -DCMAKE_SHARED_LINKER_FLAGS='-L/opt/local/libexec/openssl3/lib -Wl,-headerpad_max_install_names -latomic -L/opt/local/lib -lMacportsLegacySupport'
Default Directories:
    User Configurations ________________________________________ : /opt/local/etc/netdata
    Stock Configurations _______________________________________ : /opt/local/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /opt/local/var/cache/netdata
    Permanent Databases ________________________________________ : /opt/local/var/lib/netdata
    Plugins ____________________________________________________ : /opt/local/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /opt/local/share/netdata/web
    Log Files __________________________________________________ : /opt/local/var/log/netdata
    Lock Files _________________________________________________ : /opt/local/var/lib/netdata/lock
    Home _______________________________________________________ : /opt/local/var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : 
    Kernel Version _____________________________________________ : 
    Operating System ___________________________________________ : 
    Operating System ID ________________________________________ : 
    Operating System ID Like ___________________________________ : 
    Operating System Version ___________________________________ : 
    Operating System Version ID ________________________________ : 
    Detection __________________________________________________ : 
Hardware:
    CPU Cores __________________________________________________ : 
    CPU Frequency ______________________________________________ : 
    RAM Bytes __________________________________________________ : 
    Disk Capacity ______________________________________________ : 
    CPU Architecture ___________________________________________ : 
    Virtualization Technology __________________________________ : 
    Virtualization Detection ___________________________________ : 
Container:
    Container __________________________________________________ : 
    Container Detection ________________________________________ : 
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : 
    Container Operating System ID ______________________________ : 
    Container Operating System ID Like _________________________ : 
    Container Operating System Version _________________________ : 
    Container Operating System Version ID ______________________ : 
    Container Operating System Detection _______________________ : 
Features:
    Built For __________________________________________________ : MacOS
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip brotli)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
    Memory Allocator ___________________________________________ : system
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd lz4)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : NO
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : YES
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
    libmnl (library for working with netfilter) ________________ : NO
    stacktraces (library for getting stack traces) _____________ : backtrace
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : NO
    cgroup-network (associate interfaces to CGROUPS) ___________ : NO
    proc (monitor Linux systems) _______________________________ : NO
    tc (monitor Linux network QoS) _____________________________ : NO
    diskspace (monitor Linux mount points) _____________________ : NO
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : YES
    windows (monitor Windows systems) __________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : NO
    cups (monitor printers and print jobs) _____________________ : YES
    ebpf (monitor system calls) ________________________________ : NO
    freeipmi (monitor enterprise server H/W) ___________________ : NO
    network-viewer (monitor TCP/UDP IPv4/6 sockets) ____________ : NO
    systemd-journal (monitor journal logs) _____________________ : NO
    windows-events (monitor Windows events) ____________________ : NO
    nfacct (gather netfilter accounting) _______________________ : NO
    perf (collect kernel performance events) ___________________ : NO
    slabinfo (monitor kernel object caching) ___________________ : NO
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : NO
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
Runtime Information:
    Profile ____________________________________________________ : standalone
    Stream Parent (accept data from Children) __________________ : NO
    Stream Child (send data to a Parent) _______________________ : NO
    Total System Memory ________________________________________ : 17179869184
    Available System Memory ____________________________________ : 15250436096
```

### Additional info

```
[ 37%] Building C object CMakeFiles/netdatacli.dir/src/cli/cli.c.o
/opt/local/var/macports/build/_opt_local_ppcports_sysutils_netdata/netdata/work/netdata-v2.6.1/src/aclk/mqtt_websockets/mqtt_wss_client.c:623:9: warning: "NSEC_PER_USEC" redefined
  623 | #define NSEC_PER_USEC   1000ULL
      |         ^~~~~~~~~~~~~
In file included from /usr/include/mach/mach_types.h:98,
                 from /usr/include/mach/mach.h:66,
                 from /opt/local/include/uv/darwin.h:26,
                 from /opt/local/include/uv/unix.h:59,
                 from /opt/local/include/uv.h:71,
                 from /opt/local/var/macports/build/_opt_local_ppcports_sysutils_netdata/netdata/work/netdata-v2.6.1/src/libnetdata/common.h:101,
                 from /opt/local/var/macports/build/_opt_local_ppcports_sysutils_netdata/netdata/work/netdata-v2.6.1/src/libnetdata/libnetdata.h:10,
                 from /opt/local/var/macports/build/_opt_local_ppcports_sysutils_netdata/netdata/work/netdata-v2.6.1/src/aclk/mqtt_websockets/mqtt_wss_client.c:7:
/usr/include/mach/clock_types.h:84:9: note: this is the location of the previous definition
   84 | #define NSEC_PER_USEC   1000            /* nanoseconds per microsecond */
      |         ^~~~~~~~~~~~~
/opt/local/var/macports/build/_opt_local_ppcports_sysutils_netdata/netdata/work/netdata-v2.6.1/src/aclk/mqtt_websockets/mqtt_wss_client.c:624:9: warning: "USEC_PER_SEC" redefined
  624 | #define USEC_PER_SEC    1000000ULL
      |         ^~~~~~~~~~~~
/usr/include/mach/clock_types.h:85:9: note: this is the location of the previous definition
   85 | #define USEC_PER_SEC    1000000         /* microseconds per second */
      |         ^~~~~~~~~~~~
/opt/local/var/macports/build/_opt_local_ppcports_sysutils_netdata/netdata/work/netdata-v2.6.1/src/aclk/mqtt_websockets/mqtt_wss_client.c:626:9: warning: "NSEC_PER_SEC" redefined
  626 | #define NSEC_PER_SEC    1000000000ULL
      |         ^~~~~~~~~~~~
/usr/include/mach/clock_types.h:86:9: note: this is the location of the previous definition
   86 | #define NSEC_PER_SEC    1000000000      /* nanoseconds per second */
      |         ^~~~~~~~~~~~
```

--- a/src/aclk/mqtt_websockets/mqtt_wss_client.c
+++ b/src/aclk/mqtt_websockets/mqtt_wss_client.c
@@ -620,24 +620,6 @@ int mqtt_wss_connect(
     return 0;
 }
 
-#define NSEC_PER_USEC   1000ULL
-#define USEC_PER_SEC    1000000ULL
-#define NSEC_PER_MSEC   1000000ULL
-#define NSEC_PER_SEC    1000000000ULL
-
-static uint64_t boottime_usec(void) {
-    struct timespec ts;
-#if defined(__APPLE__) || defined(__FreeBSD__)
-    if (clock_gettime(CLOCK_MONOTONIC, &ts) == -1) {
-#else
-    if (clock_gettime(CLOCK_BOOTTIME, &ts) == -1) {
-#endif
-        nd_log(NDLS_DAEMON, NDLP_ERR, "clock_gettimte failed");
-        return 0;
-    }
-    return (uint64_t)ts.tv_sec * USEC_PER_SEC + (ts.tv_nsec % NSEC_PER_SEC) / NSEC_PER_USEC;
-}
-
 #define MWS_TIMED_OUT 1
 #define MWS_ERROR 2
 #define MWS_OK 0
@@ -656,10 +638,10 @@ static const char *mqtt_wss_error_tos(int ec)
 
 static int mqtt_wss_service_all(mqtt_wss_client client, int timeout_ms)
 {
-    uint64_t exit_by_us = boottime_usec() + (timeout_ms * NSEC_PER_MSEC);
+    uint64_t exit_by_us = now_boottime_usec() + (timeout_ms * NSEC_PER_MSEC);
     client->poll_fds[POLLFD_SOCKET].events |= POLLOUT; // TODO when entering mwtt_wss_service use out buffer size to arm POLLOUT
     while (rbuf_bytes_available(client->ws_client->buf_write)) {
-        const uint64_t now_us = boottime_usec();
+        const uint64_t now_us = now_boottime_usec();
         if (now_us >= exit_by_us)
             return MWS_TIMED_OUT;
         if (mqtt_wss_service(client, (exit_by_us - now_us) / USEC_PER_SEC))

